CMakeLists.txt

```diff
@@ -136,6 +136,10 @@ if(NOT HAVE_TRILINOS AND HAVE_KOKKOS)
     endif()
     # set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${Kokkos_CXX_FLAGS}")
     set (CMAKE_LD_FLAGS "${CMAKE_LD_FLAGS} ${Kokkos_LD_FLAGS}")
+    #
+    if (NOT NIMBLE_HAVE_MPI)
+      message(FATAL_ERROR "-- NimbleSM with Kokkos requires MPI.")
```

Comment:
Do we need this line? We should be able to build with kokkos and without MPI. Is it creating more build errors?

Reply:
At least `nimble_kokkos.cc` does not compile when `NIMBLE_HAVE_MPI` is set. Several statements need to be protected in `nimble_kokkos.cc` with `ifdef`.

Reply:
any resolution here? I'd like to merge

Reply:
FYI with this CMakeLists I can build and pass tests - so it ok for me. 
- should we treat the line 141 change as an interim or just revert it?
- please see the request to prune mpi_buckets below

Reply:
Line 141 is an interim solution until issue #22 is addressed.